pghoard/transfer.py

```diff
@@ -147,14 +147,32 @@ def untrack_upload_event(self, file_key: str) -> None:
 
     def increment(self, file_key: str, total_bytes_uploaded: float) -> None:
         metric_data = {}
+        persisted_progress = PersistedProgress.read(metrics=self.metrics)
+
         with self._tracked_events_lock:
             if file_key not in self._tracked_events:
                 raise Exception(f"UploadEvent for {file_key} is not being tracked.")
 
             file_type = self._tracked_events[file_key].file_type
             if file_type in (
-                FileType.Basebackup, FileType.Basebackup_chunk, FileType.Basebackup_delta, FileType.Basebackup_delta_chunk
+                FileType.Basebackup,
+                FileType.Basebackup_chunk,
+                FileType.Basebackup_delta,
+                FileType.Basebackup_delta_chunk,
             ):
+                progress_info = persisted_progress.get(file_key)
+                if total_bytes_uploaded > progress_info.current_progress:
+                    progress_info.update(total_bytes_uploaded)
+                    persisted_progress.write(metrics=self.metrics)
+                    self.metrics.gauge("pghoard.basebackup_stalled", 0)
```

Comment:
This may cause an increase in the storage cost of our metrics database, if we have to send 0 for every database

Reply:
yep, is it better to move to where the upload succeeded (result.success so less metric is emitted)?

Reply:
I am not sure how many files are typically uploaded this could still result in a lot data being sent. Are you able to test how many file are uploaded in a 10Gib Backup and we can do some math to figure out the cost of this metric?

We could also add a config option to only send 0 when the config `pghoard.basebackup_stalled_send_zero` is enabled. Cause this metric is really gonna only be used when  `pghoard.basebackup_stalled", 0` is greater than like 600 seconds

Reply:
as long as it's only one timeseries for each host, the cost shouldn't be significant. Markus did some research a while back showing that it's a tiny cost per data point; the cost relates to the number of timeseries.

Remember, telegraf is sending a pile of metrics every 30s. If it has one more in each 30s transmission, and it's usually zero, it's not significant. As soon as we start only sending the zeroes in certain cases, we can't use `last_over_time[24h]` or anything similar in our promql.

Reply:
see for example: https://aiven-io.slack.com/archives/D043BG00XV2/p1680501289527849